src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+        // we check if the current libs is a package
+        // if its a package
+        // get name from package
+        // if not get name from folder
+        // if its root
+        // give default name
+
+        job.start( ' for build jar check jarName ' );
+        if( jarName.len() == 0 ){
+            // jarName is empty
+            //shell.printString( ' jarName is empty ' );
+            job.addLog( ' jarName is empty ' );
+            job.addLog( ' currentLibsDir-> #currentLibsDir# ' );
+            job.addLog( ' currentProjectRoot-> #currentProjectRoot# ' );
+            if( packageService.isPackage( currentProjectRoot ) ) {
+                // its a package
+                //shell.printString( ' dir is a package ' );
+                job.addLog( ' dir is a package ' );
+                job.start( ' get jarName From Package ' );
+                jarName = getJarNameFromPackage( currentProjectRoot );
+                job.complete( getVerbose() );
+            } else {
+                // it is not a package its a normal folder
+                //shell.printString( ' dir is not a package ' );
+                job.addLog( ' dir is not a package ' );
+                var word = ListLast( getProjectRoot(), "\/" );
+                if( fileSystemUtil.isWindows() ) {
+                    //filesys is windows
+                    //shell.printString( ' filesys is windows #word# ' );
+                    job.addLog( ' filesys is windows #word# ' );
+                } else {
+                    //linux or mac
+                    //shell.printString( ' filesys is linux/mac #word# ' );
+                    job.addLog( ' filesys is linux/mac #word# ' );
+                }
+            }
+        } else {
+            // jarName is not empty
+            shell.printString( ' jarName not is empty ' );
+            job.addLog( ' jarName not is empty ' );
+        }
+        job.complete( getVerbose() );
+
+        try{
+            writeTempSourceFile( tempSrcFileName,['D:\Javatest\greetings\classes\**.class'], ".class" );
+
+            //j = 'run "#getJavaBinFolder()#jar" --file #currentLibsDir##jarName# #getJarOptionsString()#';
+            //j = 'run "#getJavaBinFolder()#jar" --create --file #currentLibsDir#testX.jar "@#tempSrcFileName#" #getJarOptionsString()#';
+            j = 'run jar --create --file "#currentLibsDir##jarName#" "@#tempSrcFileName#" #getJarOptionsString()# ';
+            //shell.printString( " " & j & " " );
+            job.addLog( " " & j & " " );
+            //command( j ).run();
+
+        } finally {
+			if ( FileExists( tempSrcFileName ) ){
+				fileDelete( tempSrcFileName );
+			}
+        }
+
+
+	}
+
+    function getJarNameFromPackage( string currentFolder ){
+        job.addLog( ' inside getJarNameFromPackage() ' );
+        var boxJSON = packageService.readPackageDescriptor( currentFolder );
+        var packageName = "";
+		var packageVersion = "";
+        if( len( Evaluate( "boxJSON.slug" ) ) != 0 ) {
```

Comment:
There's no need for `evaluate()` or ` != 0` here.  Just do
```
if( len( boxJSON.slug ) ) {
```

Reply:
done